ipatests/test_xmlrpc/test_whoami_plugin.py

```diff
@@ -0,0 +1,111 @@
+#
+# Copyright (C) 2017  FreeIPA Contributors see COPYING for license
+#
+"""
+Test the `ipaserver/plugins/whoami.py` module.
+"""
+
+import pytest
+
+from ipalib import api
+from ipatests.util import (unlock_principal_password,
+                           get_entity_keytab,
+                           change_principal,
+                           assert_deepequal,
+                           host_keytab)
+from ipatests.test_xmlrpc.xmlrpc_test import XMLRPC_test
+from ipatests.test_xmlrpc.tracker.user_plugin import UserTracker
+from ipatests.test_xmlrpc.tracker.host_plugin import HostTracker
+from ipatests.test_xmlrpc.tracker.service_plugin import ServiceTracker
+
+
+@pytest.fixture(scope='function')
+def krb_user(request):
+    tracker = UserTracker(u'krb_user', u'krb', u'test_user')
+
+    return tracker.make_fixture(request)
+
+
+@pytest.fixture
+def krb_host(request):
+    tracker = HostTracker(u'krb-host')
+
+    return tracker.make_fixture(request)
+
+
+@pytest.fixture(scope='function')
+def krb_service(request, krb_host):
+    krb_host.ensure_exists()
+
+    tracker = ServiceTracker(name=u'SERVICE', host_fqdn=krb_host.name)
+
+    return tracker.make_fixture(request)
+
+
+@pytest.mark.tier1
+class test_whoami(XMLRPC_test):
+    """
+    Test the 'whoami' plugin.
+    """
+    domain = api.env.domain
+    realm = api.env.realm
+    oldpw, newpw = u"Secret1234", u"Secret123"
+
+    def test_whoami_users(self, krb_user):
+        """
+        Testing whoami with user's tgt
+        """
+        krb_user.ensure_exists()
+
+        pwdmod = krb_user.make_update_command({'userpassword': self.oldpw})
+        pwdmod()
+
+        unlock_principal_password(krb_user.name, self.oldpw, self.newpw)
+
+        with change_principal(krb_user.name, self.newpw):
+            try:
+                result = api.Command.whoami()
+                expected = {u'object': u'user',
+                            u'command': u'user_show/1',
+                            u'arguments': (u'krb_user',)}
+                assert_deepequal(expected, result)
+            except Exception as exc:
```

Comment:
What is this supposed to do? Any exception raised during the test will be recorded as a failure. Why on top of that do you need to mask the original exception?